pkg/stanza/fileconsumer/internal/header/reader.go

```diff
@@ -19,20 +19,20 @@ import (
 var ErrEndOfHeader = errors.New("end of header")
 
 type Reader struct {
-	logger   *zap.SugaredLogger
+	logger   *zap.Logger
```

Comment:
Not necessary, but for consistency with other structs, why not store the `component.TelemetrySettings` here instead?

Reply:
Yeah, good point. Easy enough to do anyways